src/Diagnostics.ModelsAndUtils/Models/Resource/ApiManagementService.cs

```diff
@@ -22,6 +23,16 @@ public class ApiManagementService : ApiManagementServiceFilter, IResource
         /// </summary>
         public string ResourceGroup { get; set; }
 
+        /// <summary>
+        /// Resource URI
+        /// </summary>
+        public string ResourceUri {
```

Comment:
Can you abstract these out in a base class for Resource which Implements an IResource.

Reply:
actually never mind.  The resource class inherits members from filter class, so not sure if it would be possible